include/mxnet/ndarray.h

```diff
@@ -184,11 +138,17 @@ class NDArray {
           const TBlob &data, const std::vector<TBlob> &aux_data, int dev_id)
       : ptr_(std::make_shared<Chunk>(stype, data, aux_data, dev_id)), shape_(shape),
         dtype_(data.type_flag_), storage_type_(stype), entry_({nullptr, 0, 0}) {
-#if MKL_EXPERIMENTAL == 1
-    Mkl_mem_ = std::make_shared<MKLMemHolder>();
-#endif
   }
 
+  inline bool IsView() const {
+    // Sparse arrays don't have a view.
+    if (storage_type() == kRowSparseStorage || storage_type() == kCSRStorage)
```

Comment:
In case more sparse storages are added in the future, I suggest change it to `if (storage_type() != kDefaultStorage)`